src/Microsoft.TemplateEngine.Edge/Components.cs

```diff
@@ -29,7 +30,9 @@ public static class Components
                 (typeof(IInstallerFactory), new FolderInstallerFactory()),
                 (typeof(IInstallerFactory), new NuGetInstallerFactory()),
                 (typeof(ITemplateConstraintFactory), new OSConstraintFactory()),
-                (typeof(ITemplateConstraintFactory), new HostConstraintFactory())
+                (typeof(ITemplateConstraintFactory), new HostConstraintFactory()),
+                (typeof(ITemplateConstraintFactory), new WorkloadConstraintFactory()),
```

Comment:
Good point. Do you suggest additional property here in edge (e.g. somethink like `SdkSpecificComponents`)? Or anywhere else?

Reply:
Or maybe keep `AllComponents` to have all (as the name suggest) and expose a new one with more universally applicable components (e.g. `UniversalComponents`)?

Reply:
As discussed - those components are now made public (only types and default ctors), so that host developers can compose them as they wish.
Further continuation of this work is moved to separate task (as there are several dozens of components): https://github.com/dotnet/templating/issues/4767